pkg/util/test/util.go

```diff
@@ -2928,6 +2934,171 @@ func ValidatePrometheus(pxImageList map[string]string, cluster *corev1.StorageCl
 	return nil
 }
 
+func ValidateGrafana(cluster *corev1.StorageCluster) error {
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		return err
+	}
+	if opVersion.LessThanOrEqual(opVer23_8) {
+		logrus.Infof("Skipping grafana validation for operation version: [%s]", opVersion.String())
+		return nil
+	}
+
+	shouldBeInstalled := cluster.Spec.Monitoring != nil && cluster.Spec.Monitoring.Grafana != nil &&
+		cluster.Spec.Monitoring.Grafana.Enabled && cluster.Spec.Monitoring.Prometheus != nil &&
+		cluster.Spec.Monitoring.Prometheus.Enabled
+	err = ValidateGrafanaDeployment(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+	err = ValidateGrafanaService(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateGrafanaImage(shouldBeInstalled bool) error {
+	err := ValidateGrafanaDeployment(shouldBeInstalled)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateGrafanaDeployment(shouldBeInstalled bool) error {
+	check := func() (interface{}, bool, error) {
+		pods, err := coreops.Instance().GetPods("kube-system", map[string]string{"app": "grafana"})
+		if err != nil {
+			return nil, true, err
+		}
+		if shouldBeInstalled {
+			if len(pods.Items) > 0 && pods.Items[0].Status.Phase == v1.PodRunning {
+				logrus.Infof("Grafana installed successfully")
+				return "", false, nil
+			} else {
+				logrus.Errorf("Grafana not yet ready")
+				return "", true, fmt.Errorf("grafana is not installed when it should be")
+			}
+		} else {
+			if len(pods.Items) < 1 {
+				logrus.Infof("Grafana uninstalled successfully")
+				return "", false, nil
+			} else {
+				logrus.Errorf("Grafana not yet uninstalled")
+				return "", true, fmt.Errorf("grafana is installed when it is expected to be uninstalled")
+			}
+		}
+	}
+
+	if _, err := task.DoRetryWithTimeout(check, 2*time.Minute, 30*time.Second); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ValidateGrafanaDeploymentImage(image string) error {
```

Comment:
Use this function as part of your `ValidateGrafanaDeployment()`, we need to validate images as a part of deployment validation, see example in the `ValidateStork()` or other components validation functions, also take a look at this function that does image validation inside containers https://github.com/libopenstorage/operator/blob/e4f90a46c0a2e28416d5acf44e953a5c42398010/pkg/util/test/util.go#L2577, this works with custom registries and trims image names accordingly using operator function `GetImageURN()`
NOTE: You need to validate images when they are part of the version image list https://install.portworx.com/3.0.0/version?kbver=1.24.0